### PR TITLE
Add structured clone tests for errors

### PR DESCRIPTION
**Pull request by yutakahirano**
This is tests for spec changes.
 - https://github.com/whatwg/html/pull/4665
 - https://github.com/heycam/webidl/pull/732

### CODE REVIEW

File: html/infrastructure/safe-passing-of-structured-data/structuredclone_0.html [hunk 1]
```diff
@@ -367,6 +367,154 @@
...
+                    const error = URIError("some message");
+                    error.foo = "bar";
+                    worker.postMessage(error);
+                });
+            },
+            function() {
+                var t = async_test("Cloning a modified Error");
+                t.id = 34;
+                worker.onmessage = t.step_func_done(function(e) {
+                    assert_equals(Object.getPrototypeOf(e.data), URIError.prototype, "Checking prototype");
```
[the remark this answers is not in the document]
Comment (yutakahirano) [reply html/infrastructure/safe-passing-of-structured-data/structuredclone_0.html]: You're right, thanks! Fixed.

File: html/infrastructure/safe-passing-of-structured-data/structuredclone_0.html [hunk 2]
```diff
@@ -367,6 +367,154 @@
...
+                worker.onmessage = t.step_func_done(function(e) {
+                    assert_equals(Object.getPrototypeOf(e.data), DOMException.prototype, "Checking prototype");
+                    assert_equals(e.data.constructor, DOMException, "Checking constructor");
+                    assert_equals(e.data.name, "IndexSizeError", "Checking name");
+                    assert_equals(e.data.message, "some message", "Checking message");
+                    assert_equals(e.data.code, 1, "Checking code");
+                    assert_equals(e.data.foo, undefined, "Checking custom property");
+                });
+                t.step(function() {
+                    const error = new DOMException("some message", "IndexSizeError");
```
Comment (TimothyGu) [inline html/infrastructure/safe-passing-of-structured-data/structuredclone_0.html]: Can we test also on some error thrown by DOM functions?

Comment (yutakahirano) [reply html/infrastructure/safe-passing-of-structured-data/structuredclone_0.html]: Done.

**Review by TimothyGu — APPROVED** [review]

File: html/infrastructure/safe-passing-of-structured-data/structuredclone_0.html [hunk 3]
```diff
@@ -367,6 +367,173 @@
...
+                t.step(function() {
+                    const error = URIError("some message");
+                    Object.setPrototypeOf(error, SyntaxError.prototype);
+                    error.message = {toString: () => "another message" }
+                    error.constructor = RangeError;
+                    error.name = "TypeError";
+                    error.foo = "bar";
+                    worker.postMessage(error);
+                });
+            },
```
Comment (domenic) [inline html/infrastructure/safe-passing-of-structured-data/structuredclone_0.html]: Can you add tests that:

- If you set message to a getter that returns a string (via Object.defineProperty), it does not get serialized, and on the other side, `err.hasOwnProperty("message")` is false
- If you construct an error with no message (e.g. `new Error()` or `new EvalError()`, then on both sides, `err.hasOwnProperty("message")` is false?

**Review by domenic — APPROVED** [review]
LGTM with additional test suggestions. I can work on an optional test for stack.

File: html/infrastructure/safe-passing-of-structured-data/structuredclone_0.html (hunk 3 above)
Comment (yutakahirano) [reply html/infrastructure/safe-passing-of-structured-data/structuredclone_0.html]: Done.

**Comment by domenic** [comment]
Added another test, PTAL @yutakahirano

**Comment by domenic** [comment]
Going to close and reopen to try to fix the taskcluster failure.

File: html/infrastructure/safe-passing-of-structured-data/structuredclone_0.html [hunk 4]
```diff
@@ -367,7 +368,240 @@
...
+                t.id = 35;
+                worker.onmessage = t.step_func_done(function(e) {
+                    assert_equals(Object.getPrototypeOf(e.data), Error.prototype, "Checking prototype");
+                    assert_equals(e.data.constructor, Error, "Checking constructor");
+                    assert_equals(e.data.name, "Error", "Checking name");
+                    assert_equals(e.data.message, "some message", "Checking message");
+                    assert_equals(e.data.foo, undefined, "Checking custom property");
+                });
+                t.step(function() {
+                    const error = frames[0].URIError("some message");
```
Comment (yutakahirano) [inline html/infrastructure/safe-passing-of-structured-data/structuredclone_0.html]: Can you check Object.getPrototypeof(error), error.constructor and error.name?

Comment (domenic) [reply html/infrastructure/safe-passing-of-structured-data/structuredclone_0.html]: That shouldn't be necessary in this setup step, I think? But I will add them anyway, and we can always revert.

Comment (yutakahirano) [reply html/infrastructure/safe-passing-of-structured-data/structuredclone_0.html]: Thanks!

**Comment by foolip** [comment]
Rebasing this should fix the Taskcluster failure, while just running again doesn't because the Taskcluster config from the PR branch is used and not what is on master.

But admin merging instead since the failure is a known and fixed one.

**Comment by Brettski19871987** [comment]
Verafacation for my wallet

**Pull request merged**